[MacOS-12.2.1][x86_64] make test_bootstrap Error

Hello,  nopforth failed on MacOS

compile src/x86_64/Darwin.s failed
```
mkdir -p bin
cc -ggdb -Isrc -c -o bin/nop.o src/x86_64/Darwin.s
src/x86_64/Darwin.s:5:10: error: Could not find include file 'boot.s'

....
src/x86_64/Darwin.s:256:9: error: Could not find incbin file 'arch.ns'
.incbin "arch.ns"

```

after change that 2 lines to
```
.include "x86_64/boot.s"
.incbin "x86_64/arch.ns"
```
it compiles, but test failed
```
mkdir -p bin
cc -ggdb -Isrc -c -o bin/nop.o src/x86_64/Darwin.s
cc -ggdb -o bin/nop bin/nop.o 
/bin/sh: line 1: 10066 Done                    echo
     10067 Segmentation fault: 11  | bin/nop
make: *** [test_bootstrap] Error 139

```

--- a/.github/workflows/tests-linux-x86_64.yml
+++ b/.github/workflows/tests-linux-x86_64.yml
@@ -1,4 +1,4 @@
-name: Run tests (linux/x86_64)
+name: Linux/x86_64
 on:
   pull_request:
     branches:

--- a/.github/workflows/tests-linux-x86_64.yml
+++ b/.github/workflows/tests-linux-x86_64.yml
@@ -1,4 +1,4 @@
-name: Run tests
+name: Run tests (linux/x86_64)
 on:
   pull_request:
     branches:

--- a/.github/workflows/tests-macos-x86_64.yml
+++ b/.github/workflows/tests-macos-x86_64.yml
@@ -1,0 +1,19 @@
+name: Run tests (macos/x86_64)
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+jobs:
+  build:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Build
+        run: make
+
+      - name: Test
+        run: make test

--- a/.github/workflows/tests-macos-x86_64.yml
+++ b/.github/workflows/tests-macos-x86_64.yml
@@ -1,4 +1,4 @@
-name: Run tests (macos/x86_64)
+name: macOS/x86_64
 on:
   pull_request:
     branches:

--- a/src/dicts.s
+++ b/src/dicts.s
@@ -594,6 +594,7 @@ Br_header:
     .byte 2
     .ascii "Br"
 
+    .align 8
 _flatest: .quad Br_header
 
 #
@@ -858,4 +859,5 @@ clit_header:
     .byte 3
     .ascii "lit"
 
+    .align 8
 _mlatest: .quad clit_header

--- a/src/dicts.s
+++ b/src/dicts.s
@@ -83,84 +83,84 @@ sysdlclose_header:
     .byte 7
     .ascii "dlclose"
 
-   .align 8
+    .align 8
 sysdlerror_header:
     .quad sysdlclose_header
     .quad sysdlerror
     .byte 7
     .ascii "dlerror"
 
-   .align 8
+    .align 8
 sysdlsym_header:
     .quad sysdlerror_header
     .quad sysdlsym
     .byte 5
     .ascii "dlsym"
 
-   .align 8
+    .align 8
 sysdlopen_header:
     .quad sysdlsym_header
     .quad sysdlopen
     .byte 6
     .ascii "dlopen"
 
-   .align 8
+    .align 8
 sysalloc_header:
     .quad sysdlopen_header
     .quad sysalloc
     .byte 8
     .ascii "sysalloc"
 
-   .align 8
+    .align 8
 sysresize_header:
     .quad sysalloc_header
     .quad sysresize
     .byte 9
     .ascii "sysresize"
 
-   .align 8
+    .align 8
 sysfree_header:
     .quad sysresize_header
     .quad sysfree
     .byte 7
     .ascii "sysfree"
 
-   .align 8
+    .align 8
 syscall6_header:
     .quad sysfree_header
     .quad syscall6
     .byte 8
     .ascii "syscall6"
 
-   .align 8
+    .align 8
 syscall5_header:
     .quad syscall6_header
     .quad syscall5
     .byte 8
     .ascii "syscall5"
 
-   .align 8
+    .align 8
 syscall4_header:
     .quad syscall5_header
     .quad syscall4
     .byte 8
     .ascii "syscall4"
 
-   .align 8
+    .align 8
 syscall3_header:
     .quad syscall4_header
     .quad syscall3
     .byte 8
     .ascii "syscall3"
 
-   .align 8
+    .align 8
 syscall2_header:
     .quad syscall3_header
     .quad syscall2
     .byte 8
     .ascii "syscall2"
 
-   .align 8
+    .align 8
 syscall1_header:
     .quad syscall2_header
     .quad syscall1
@@ -704,6 +704,7 @@ cfetch_header:
     .byte 1
     .ascii "@"
 
+    .align 8
 cfetchplus_header:
     .quad cfetch_header
     .quad cfetchplus
@@ -717,6 +718,7 @@ c1fetch_header:
     .byte 2
     .ascii "1@"
 
+    .align 8
 c1fetchplus_header:
     .quad c1fetch_header
     .quad c1fetchplus

--- a/src/x86_64/Darwin.s
+++ b/src/x86_64/Darwin.s
@@ -9,7 +9,9 @@
 .equ PROT_EXEC,     0x4
 .equ MAP_SHARED,    0x1
 .equ MAP_PRIVATE,   0x2
-.equ MAP_ANONYMOUS, 0x1000
+.equ MAP_JIT,       0x0800
+.equ MAP_ANON,      0x1000
+
 
 # System V x86-64 ABI requires a stack aligned on a 16 bytes boundary.
 # The prolog saves the return stack pointer on rbx since the latter must be
@@ -230,23 +232,40 @@ setupenv:
     mov %rcx, _args(%rip)
     ret
 
+    .data
+_dictstart: .quad 0
+_dictsize = 0x100000
+_codestart: .quad 0
+_codesize = 0x100000
+
+    .text
 resetdict:
+    # dictionary
     dup_
-    mov $0, %rdi         # addr
-    mov $0x100000, %rsi  # length
-    mov $(PROT_READ | PROT_WRITE | PROT_EXEC), %rdx  # prot
-    mov $(MAP_ANONYMOUS | MAP_SHARED), %rcx          # flags
-    mov $-1, %r8         # fd
-    mov $0, %r9          # offset (ignored)
-    call _mmap
+    mov $_dictsize, %rdi
+    call _malloc
     test %rax, %rax
     jz 1f
     mov %rax, _h(%rip)
+    mov %rax, _dictstart(%rip)
+
+    # code space
+    mov $0, %rdi          # addr
+    mov $_codesize, %rsi  # length
+    mov $(PROT_READ | PROT_WRITE | PROT_EXEC), %rdx  # prot
+    mov $(MAP_ANON | MAP_PRIVATE | MAP_JIT), %rcx    # flags
+    mov $-1, %r8          # fd
+    mov $0, %r9           # offset (ignored)
+    call _mmap
+    cmp $-1, %rax         # MAP_FAILED?
+    je 1f
+    mov %rax, codepp(%rip)
+    mov %rax, _codestart(%rip)
     drop_
     ret
 1:
     mov $1, %rax
-    call sysexit
+    jmp sysexit
 
 .data
 .include "dicts.s"

--- a/src/x86_64/Darwin.s
+++ b/src/x86_64/Darwin.s
@@ -2,7 +2,7 @@
 # Copyright (c) 2018-2022 Iruatã Martins dos Santos Souza
 
     .text
-.include "boot.s"
+.include "x86_64/boot.s"
 
 .equ PROT_READ,     0x1
 .equ PROT_WRITE,    0x2
@@ -253,7 +253,7 @@ resetdict:
 
 _kernbuf:
 .incbin "comments.ns"
-.incbin "arch.ns"
+.incbin "x86_64/arch.ns"
 .incbin "flowcontrol.ns"
 .incbin "interactive.ns"
 .incbin "dictionary.ns"

--- a/src/x86_64/arch.ns
+++ b/src/x86_64/arch.ns
@@ -69,8 +69,5 @@ forth
 : branch? ( a -> bool )
    1@ $E9 = ;
 
-: branch-offset-multiplier ( -> a )
-   [ here 0 , ] lit ;
-
 : NOPARCH ( -> a u )
    [ 6 here  $78 1, $38 1, $36 1, $5F 1, $36 1, $34 1, ] lit lit ;

--- a/src/x86_64/boot.s
+++ b/src/x86_64/boot.s
@@ -25,6 +25,7 @@ type:  # a u ->
     ret
 
     .data
+    .align 8
 _keyxt: .quad 0
     .text
 keyxt:  # -> a
@@ -190,6 +191,7 @@ number:  # a u -> n err
     ret
 
     .data
+    .align 8
 _infd:   .quad 0    # file descriptor feeding the input
 _inbuf:  .quad 0    # pointer to input buffer
 _intot:  .quad 0    # input buffer total size
@@ -290,6 +292,7 @@ parse:  # delim -> a u
     ret
 
     .data
+    .align 8
 codepp: .quad 0  # next code address
 _h:    .quad 0   # next dictionary address
 _latest: .quad _flatest
@@ -865,6 +868,7 @@ clit:
     jmp cinstr
 
     .data
+    .align 8
 _abortxt:  .quad abort1
 _qmsg:     .ascii "?\n"
 _qlen =    . - _qmsg
@@ -894,6 +898,7 @@ abort:
     jmp *_abortxt(%rip)
 
     .data
+    .align 8
 _search:
   .quad 0          # 1st dictionary to be searched
   .quad 0          # 2nd dictionary to be searched
@@ -1003,6 +1008,7 @@ bye:
     call sysexit
 
     .data
+    .align 8
 dstack: .space 8192, 0
 dstack0: .quad 0
 
@@ -1060,6 +1066,7 @@ resetstacks:
     ret
 
     .data
+    .align 8
 _undererr: .ascii "stack underflow!\n"
 _undererrlen = . - _undererr
 _overerr:  .ascii "stack overflow!\n"

